src/haz3lcore/lang/term/TermBase.re

```diff
@@ -280,6 +293,10 @@ and Exp: {
     | (Parens(x), _) => fast_equal(x, e2)
     | (_, DynamicErrorHole(x, _))
     | (_, Parens(x)) => fast_equal(e1, x)
+    /* Below is kind of a hack to make EvalResult.calculate go after adding a projector.
+     * We should clarify syntactic/semantic equality here */
+    | (Probe(x1, _), Probe(x2, _)) => fast_equal(x1, x2)
+    | (Probe(_, _), _) => false
```

Comment:
I just added https://github.com/hazelgrove/hazel/issues/1563. Feel free to add comments on any probe specifics.